tests/vault.test.ts

```diff
@@ -222,6 +237,67 @@ describe("Vault", () => {
 				"Assets should be the amount of assets requested"
 			).to.equal(ethers.utils.parseUnits("500", underlyingDecimals));
 		});
+
+		it("Should not allow user to withdraw before the lock up period ends", async () => {
+			const amount = ethers.utils.parseUnits("1000", underlyingDecimals);
+			await underlying.connect(alice).approve(vault.address, amount);
+
+			const latestBlockNumber = await ethers.provider.getBlockNumber();
+			const latestBlock = await ethers.provider.getBlock(latestBlockNumber);
+			await vault.connect(alice).deposit(amount, alice.address);
+
+			const lockedTime = await (
+				await vault.lockedTime(alice.address)
+			).toNumber();
+
+			expect(lockedTime).to.equal(
+				latestBlock.timestamp + Number(process.env.VAULT_LOCK_TIME) + 1
+			);
+			await expect(
+				vault
+					.connect(alice)
+					.withdraw(
+						ethers.utils.parseUnits("100", underlyingDecimals),
+						alice.address,
+						alice.address
+					)
+			).to.be.revertedWith("withdraw: Locked time not passed");
+
+			// Move past the lock up period
+			await hre.network.provider.request({
+				method: "evm_setNextBlockTimestamp",
+				params: [lockedTime + 1]
+			});
```

Comment:
might be worth making this a util if there are going to be lots of time dependant tests